### PR TITLE
[CALCITE-4844] IN-list that references columns is wrongly converted to Values, and gives incorrect results

### PR DESCRIPTION
**Pull request by NobiGo**
…o Values, and gives incorrect results

### CODE REVIEW

File: testkit/src/main/java/org/apache/calcite/test/QuidemTest.java [hunk 1]
```diff
@@ -150,6 +151,10 @@ protected void checkRun(String path) throws Exception {
                   && (Boolean) value;
               closer.add(Prepare.THREAD_EXPAND.push(b));
             }
+            if (propertyName.equals("insubquerythreshold")) {
```
Comment (zinking) [inline testkit/src/main/java/org/apache/calcite/test/QuidemTest.java:154]: is this temporary ? surprised code gen part follows naturely and requires no change.

Comment (NobiGo) [reply testkit/src/main/java/org/apache/calcite/test/QuidemTest.java:154]: No. This can be used in the .iq file to set the insubquerythreshold property.

Comment (NobiGo) [reply testkit/src/main/java/org/apache/calcite/test/QuidemTest.java:154]: `!set insubquerythreshold 0`

**Review by chunweilei — COMMENTED** [review]
LGTM

File: testkit/src/main/java/org/apache/calcite/test/QuidemTest.java (hunk 1 above)
Comment (NobiGo) [reply testkit/src/main/java/org/apache/calcite/test/QuidemTest.java:154]: @zinking Please review again, Then we can merge.

**Pull request merged**